x-pack/plugin/security/src/internalClusterTest/java/org/elasticsearch/xpack/security/authc/ApiKeyIntegTests.java

```diff
@@ -2651,15 +2650,14 @@ public void testUpdateApiKeysClearsApiKeyDocCache() throws Exception {
     }
 
     private List<RoleDescriptor> randomRoleDescriptors() {
-        boolean allowRemoteIndices = TcpTransport.isUntrustedRemoteClusterEnabled();
         int caseNo = randomIntBetween(0, 3);
         return switch (caseNo) {
             case 0 -> List.of(new RoleDescriptor(randomAlphaOfLength(10), new String[] { "all" }, null, null));
             case 1 -> List.of(
                 new RoleDescriptor(randomAlphaOfLength(10), new String[] { "all" }, null, null),
                 randomValueOtherThanMany(
                     rd -> RoleDescriptorRequestValidator.validate(rd) != null,
-                    () -> RoleDescriptorTests.randomRoleDescriptor(false, allowRemoteIndices, false)
+                    () -> RoleDescriptorTests.randomRoleDescriptor(false, true, false)
```

Comment:
Not for this PR, but we could do some more targeted clean up -- I don't think there's a need to keep this argument around anymore. Happy to raise some clean up PRs from my end. Again, not for this PR, to keep the actual FF removal as mechanical as possible.

Reply:
Yeah I'll leave this as is for now. If you search `allowRemoteIndices`, there are a few other places we do similar conditional logic. They are all good candidate for a follow-up.